src/core/src/pass/serialize.cpp

```diff
@@ -1262,77 +1262,70 @@ bool pass::Serialize::run_on_model(const std::shared_ptr<ov::Model>& model) {
     for (auto& node : model->get_ops())
         if (fp16_compression_is_disabled(node))
             disable_fp16_compression(node);
-
-    if (m_xmlFile && m_binFile) {
-        serializeFunc(*m_xmlFile, *m_binFile, model, m_version);
-    } else {
-#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
-        const auto& xmlPath_ref = ov::util::string_to_wstring(m_xmlPath);
-        const auto& binPath_ref = ov::util::string_to_wstring(m_binPath);
-        std::string message_bin = "Can't open bin file.";
-        std::string message_xml = "Can't open xml file.";
+            
+ #if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+    const auto& xmlPath_ref = ov::util::string_to_wstring(m_xmlPath);
+    const auto& binPath_ref = ov::util::string_to_wstring(m_binPath);
+    std::string message_bin = "Can't open bin file.";
+    std::string message_xml = "Can't open xml file.";
 #else
-        const auto& xmlPath_ref = m_xmlPath;
-        const auto& binPath_ref = m_binPath;
-        std::string message_bin = "Can't open bin file: \"" + binPath_ref + "\"";
-        std::string message_xml = "Can't open xml file: \"" + xmlPath_ref + "\"";
+    const auto& xmlPath_ref = m_xmlPath;
```

Comment:
Consider change:
`const auto xml_path = fs::path(m_xmlPath)`

as this conversion will be applied on `ov::util::get_directory` call.

Then some #ifdef could be removed.
The error message can be unified also, as `path` can be converted to string from native representation.